x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authc/AuthenticationTests.java

```diff
@@ -43,14 +43,23 @@ public class AuthenticationTests extends ESTestCase {
     public void testWillGetLookedUpByWhenItExists() {
         final RealmRef authenticatedBy = new RealmRef("auth_by", "auth_by_type", "node");
         final RealmRef lookedUpBy = new RealmRef("lookup_by", "lookup_by_type", "node");
-        final Authentication authentication = new Authentication(new User("user"), authenticatedBy, lookedUpBy);
+        final Authentication authentication = AuthenticationTestHelper.builder()
+            .user(new User("not-user"))
+            .realmRef(authenticatedBy)
+            .runAs()
+            .user(new User("user"))
+            .realmRef(lookedUpBy)
+            .build();
```

Comment:
The authentication object was created incorrectly: The user does not have a run-as but the authentication has a non-null lookup realm. So the new code fixed it by adding an authenticating user as I believe the intention here was to have run-as. 

The similar issue existed in a few other places and they have been fixed similarly.